### PR TITLE
Write down some docs

### PR DESCRIPTION
**Pull request by malukenho**
Linking #5

### CODE REVIEW

File: docs/Installing.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Installing
+
+Do you have to use `composer` to install this library. If you don't know what composer is, please take a look at
```
Comment (Ocramius) [inline docs/Installing.md]: `Please refer to the official composer site to install composer`

File: docs/Installing.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+[composer official site](https://getcomposer.org).
+
+With all properly installed you have to run the following command to start to be strict.
```
Comment (Ocramius) [inline docs/Installing.md]: Skip this sentence

File: docs/Installing.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+```sh
+composer require roave/strict-php
+```
+
+That's pretty much it.
```
Comment (Ocramius) [inline docs/Installing.md]: Skip or link further steps

File: docs/basic-strict.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Basic Strict
```
Comment (Ocramius) [inline docs/basic-strict.md]: Basic strict property typing

File: docs/basic-strict.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+The `StrictPhp` uses compatible with the PhpDocumentor annotation. Then you probably should be familiar with them.
+
+Take a look at following class.
+
+```php
+class Sushi
+{
+    /**
+     * @var int
```
Comment (Ocramius) [inline docs/basic-strict.md]: Probably a float ;-)

File: docs/basic-strict.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# Basic Strict
+
+With a configured bootstrap, we use notes on the properties to make your code work strictly.
```
Comment (Ocramius) [inline docs/basic-strict.md]: s/`notes`/`annotations/docblocks`
s/`to make your code work strictly`/`to make your app operate in strict mode`

File: docs/basic-strict.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+class Sushi
+{
+    /**
+     * @var int
+     */
+    public $price;
+}
+```
+
+We're talking to `StrictPhp` we can only set a integer value for the `Sushi#price` property.
```
Comment (Ocramius) [inline docs/basic-strict.md]: Not sure what `We're talking` means here

File: docs/basic-strict.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+Let's try set a integer value.
+
+```php
+$sushi = new Sushi;
+$sushi->price = 2;
+```
+
+That's work perfectly with the correct data.
```
Comment (Ocramius) [inline docs/basic-strict.md]: That works perfectly fine.

File: docs/basic-strict.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+Let's try pass a wrong data type for the property.
```
Comment (Ocramius) [inline docs/basic-strict.md]: s/pass/assigning
s/for/to

File: docs/basic-strict.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+```php
+$sushi = new Sushi;
+$sushi->price = '2';
+```
+
+We have got a:
```
Comment (Ocramius) [inline docs/basic-strict.md]: We will get an exception:

File: docs/basic-strict.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+Fatal error: Uncaught exception 'ErrorException' ...
+```
+
+You're using the most basic common functionality of `StrictPhp`.
```
Comment (Ocramius) [inline docs/basic-strict.md]: This is the most basic functionality of `StrictPhp`

File: docs/basic-strict.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+Great!
```
Comment (Ocramius) [inline docs/basic-strict.md]: Skip. Consider adding a section on array types, such as `Foo[]` and `Bar[][][]` or such ;-)

File: docs/bootstrap.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# Being Strict
```
Comment (Ocramius) [inline docs/bootstrap.md]: Enabling StrictPhp

File: docs/bootstrap.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+Start being strict is very easy, at first we just need to configure an instance of `\StrictPhp\StrictPhpKernel`
```
Comment (Ocramius) [inline docs/bootstrap.md]: To bootstrap StrictPhp, we initialize the `StrictPhp\StrictPhpKernel` singleton:

File: docs/bootstrap.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+configured properly.
+
+```php
+\StrictPhp\StrictPhpKernel::getInstance()->init([
+    'debug'        => true,
+    'cacheDir'     => sys_get_temp_dir(),
```
Comment (Ocramius) [inline docs/bootstrap.md]: Add comments about what all these properties mean

File: docs/bootstrap.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+    'includePaths' => [
+        __DIR__ . '/path/to/your/sources',
+    ],
+]);
+```
+
+More details about the options settings that can be passed to the `init` can be seen in the documentation of the
```
Comment (Ocramius) [inline docs/bootstrap.md]: More supported configuration keys can be found in the GO AOP PHP documentation (link to the right page in the docs here ;-) )

File: docs/bootstrap.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+[Go! AOP Framework](https://github.com/lisachenko/go-aop-php)
+
+It's all you need to start being Strict!
```
Comment (Ocramius) [inline docs/bootstrap.md]: Link to the next chapter instead

File: docs/basic-strict.md [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+class Sushi
+{
+    /**
+     * @var float
+     */
+    public $price;
+}
+```
+
+In that way, `StrictPhp` knows we can only set a **float* value for the `Sushi#price` property.
```
Comment (Ocramius) [inline docs/basic-strict.md]: Markdown broken here (closing * )

File: docs/immutability.md [hunk 19]
```diff
@@ -0,0 +1,36 @@
+# Immutability
+
+We also provided a `@immutable` annotation, who works pretty much a constant, ideal to use on values objects, not mutable.
```
Comment (Ocramius) [inline docs/immutability.md]: s/who/which

Comment (Ocramius) [inline docs/immutability.md]: s/a constant/as a marker for final values

File: docs/immutability.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+
+A property setted as immutable can receive a value only a single time. Your value never can be changed.
```
Comment (Ocramius) [inline docs/immutability.md]: s/setted/marked
s/Your value never can be changed/The value can never be overwritten

File: docs/immutability.md [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+The following code...
```
Comment (Ocramius) [inline docs/immutability.md]: I would skip the following example

File: docs/immutability.md [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function getPrice()
+    {
+        return $this->price;
+    }
+}
+```
+
+Can be expressed using `StrictPhp` as th follow:
```
Comment (Ocramius) [inline docs/immutability.md]: the?

File: docs/immutability.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+
+```php
+class Sushi
```
Comment (Ocramius) [inline docs/immutability.md]: Add an example of a failure, please

File: docs/Installing.md [hunk 24]
```diff
@@ -0,0 +1,9 @@
+# Installing
+
+Do you have to use `[composer](https://getcomposer.org)` to install this library.
+
+```sh
+composer require roave/strict-php
+```
+
+**Next step:** [Enabling StrictPhp](bootstrap.md)
```
Comment (Ocramius) [inline docs/Installing.md]: Add notes about the fact that `strict-php` is only available with `"minimum-stability": "dev"` right now

File: docs/basic-strict.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
+# Basic strict property typing
+
+With a configured bootstrap, we use annotations on the properties to make your app operate in strict mode.
+
+The `StrictPhp` uses compatible with the PhpDocumentor annotation. Then you probably should be familiar with them.
```
Comment (Ocramius) [inline docs/basic-strict.md]: No "The"

Comment (Ocramius) [inline docs/basic-strict.md]: "`StrictPhp` uses PhpDocumentor compatible annotations" (link the annotations doc of phpdocumentor, please).

Comment (Ocramius) [inline docs/basic-strict.md]: Remove "Then you probably should be familiar with them."

File: docs/basic-strict.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+Take a look at following class.
```
Comment (Ocramius) [inline docs/basic-strict.md]: the following

File: docs/basic-strict.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+class Sushi
+{
+    /**
+     * @var float
+     */
+    public $price;
+}
+```
+
+In that way, `StrictPhp` knows we can only set a **float** value for the `Sushi#price` property.
```
Comment (Ocramius) [inline docs/basic-strict.md]: In this way

Comment (Ocramius) [inline docs/basic-strict.md]: that we can

File: docs/basic-strict.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+
+Let's try set a integer value.
```
Comment (Ocramius) [inline docs/basic-strict.md]: setting

**Comment by Ocramius** [comment]
@malukenho since the docs are already available on the homepage, what do you think of using couscous here?

**Comment by Ocramius** [comment]
@malukenho I added the token in Roave/StrictPhp@bdf1311077a5791b6e79ea7fee565a8ee5b4706f

**Comment by Ocramius** [comment]
@malukenho can you rebase this when you got time?

**Comment by malukenho** [comment]
@Ocramius rebased

**Comment by Ocramius** [comment]
\o/

**Comment by Ocramius** [comment]
@malukenho @mnapoli weird, reports `This version of the documentation is already deployed`, but the `gh-pages` branch is not there at all :-\

**Comment by mnapoli** [comment]
Weird, did you generate manually afterwards? (I see you committed on the `gh-pages` branch) Maybe it fails when the branch doesn't exist :/ Is it alright now?

**Comment by malukenho** [comment]
@mnapoli yah. I guess the problem was raised because the branch don't exists. Now It's okay. Thanks.

> The sidebar menu is broken when we have a large numbers of menu items, some items are not displayed, (Using the default theme)

**Comment by mnapoli** [comment]
FYI

> The sidebar menu is broken when we have a large numbers of menu items, some items are not displayed, (Using the default theme)

The default theme is pretty basic: the left menu is "affixed" (fixed, not scrollable), which means if your menu spans beyond the height of the screen some items will be hidden. This is a design choice because it's supposed to help get started (better usability for small docs/project, better). For larger docs/projects it's not the best fit indeed.

**Comment by malukenho** [comment]
@mnapoli thanks for the information.

**Pull request merged**